code/processes/chainedtp.q

```diff
@@ -196,13 +185,15 @@ if[not `upd in key `.; upd:.ctp.upd];
 /- pubsub must be initialised sooner to enable tickerplant replay publishing to work
 .ps.initialise[];                                                                   
 
-/- check if the tickerplant has connected, blocks the process until a connection is established
-.ctp.initialise[];
-while[.ctp.notpconnected[];
-  /- while not connected make the process sleep for X seconds and then run the subscribe function again
-  .os.sleep[.ctp.tpconnsleep];
-  /- start chained tickerplant
-  .ctp.initialise[]];
+/- check if tickerplant is available and if not exit with error 
+.servers.startupdepnamecycles[.ctp.tickerplantname;.ctp.tpconnsleep;.ctp.tpcheckcycles];
+
+/- subscribe to tickerplant
+.ctp.subscribe[]; 
+
+/- add subscribed table schemas to .ctp.tableschemas, used in cleartables
+if[not .ctp.notpconnected[];
```

Comment:
Is it ever possible for this condition to be false? If it was false, should the CTP just exit?

Reply:
I think the condition can be removed because .servers.startupdepnamecycles ensures that it connects to the correct tickerplant, so it'll always be able to subscribe which means the condition will always be true? unless there's a way subscribing to a tickerplant can fail even though there's a connection?

Reply:
ok cool, can you remove it then please?